torch/_subclasses/fake_tensor.py

```diff
@@ -324,25 +341,28 @@ def __new__(cls, fake_mode, elem, device):
         )
 
     def __init__(self, fake_mode, elem, device: Union[torch.device, str]):
-        # elem does not need to be recorded, because FakeTensor *is a* elem
-        assert elem.device.type == "meta", elem
+        assert elem.device.type == "meta", elem.device.type
         device = device if isinstance(device, torch.device) else torch.device(device)
+        # NB: it is fine, if a little confusing, for device to be meta
```

Comment:
Hmm, what do you think about making allowing this a property of the FakeTensorMode ? Ive had a few instances where these assertions have caught bugs.

Reply:
Yeah, I'm cool with that. Can add it.